src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -380,6 +401,51 @@ def get_name(self) -> str:
         # this parameter does not need validation
         return name
 
+    # pylint: disable=unused-argument
+    def _get_location(self, read_only: bool = False, **kwargs) -> Union[str, None]:
+        """Internal function to dynamically obtain the value of location according to the context.
+
+        When location is not assigned, dynamic completion will be triggerd. Function "_get_rg_location" will be called
+        to get the location of the provided resource group, which internally used ResourceManagementClient to send
+        the request.
+
+        This function supports the option of read_only. When enabled, it will skip dynamic completion and validation.
+
+        :return: string or None
+        """
+        # read the original value passed by the command
+        location = self.raw_param.get("location")
+        # try to read the property value corresponding to the parameter from the `mc` object
+        read_from_mc = False
+        if self.mc and self.mc.location is not None:
+            location = self.mc.location
+            read_from_mc = True
+
+        # skip dynamic completion & validation if option read_only is specified
+        if read_only:
+            return location
+
+        # dynamic completion
+        if not read_from_mc and location is None:
+            location = _get_rg_location(
```

Comment:
what if failed to get the location?

Reply:
In function `_get_rg_location`, we actually call ResourceManagementClient to get the location "property" of the provided resource group (which is a required parameter). If the operation failed, it should be handled properly (raise an exception) by ResourceManagementClient.